src/dgrid/diff.ts

```diff
@@ -0,0 +1,28 @@
+import { Column, ColumnSpec } from 'dgrid/Grid';
+import { PropertyChangeRecord } from '@dojo/framework/widget-core/interfaces';
+import { auto } from '@dojo/framework/widget-core/diff';
+
+export function columnsDiff(prevColumnSpec: ColumnSpec, newColumnSpec: ColumnSpec): PropertyChangeRecord {
+	let changed = true;
+	const prevIsArray = Array.isArray(prevColumnSpec);
```

Comment:
If `prevColumnSpec` and `newColumnSpec` can be an array then have we got the diff signature typings wrong?

Reply:
I think the types are right. `ColumnSpec` can be an object or an array. Am I missing something?

Reply:
Nope I was, I thought that it might just be a type that unioned an array and non array after input the comment. But didn’t get round to checking. Thanks!